storage/poolmanager/poolmanager.go

```diff
@@ -78,6 +78,14 @@ func (pm *poolManager) Delete(name string) error {
 	return errors.Annotatef(err, "deleting pool %q", name)
 }
 
+// Replace is defined on PoolManager interface.
+func (pm *poolManager) Replace(name string, attrs map[string]interface{}) error {
+	if name == "" {
+		return MissingNameError
+	}
+	return pm.settings.ReplaceSettings(globalKey(name), attrs)
```

Comment:
This will break stuff - see Create()
We need to ensure the settings have the name and type added.
Also, where's the validation?
The only difference between create and replace is the existence of a previous pool so we need to refactor the methods to use common code.
On that basis, Replace should probably also support updating the type.

Reply:
These issue would have been picked p if there were tests added to poolmanager_test.go :-)